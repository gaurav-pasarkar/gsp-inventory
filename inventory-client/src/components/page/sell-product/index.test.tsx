import matchSnapshot from "../../../test-utils/match-snapshot";
import SellProduct from "./index";
import {render, waitFor} from "@testing-library/react";
import {act} from "react-test-renderer";

describe('Sell products', () => {
  it('should render', () => {
    matchSnapshot(<SellProduct />);
  });

  it('should render Add product by default', async () => {
    await act(async () => {
      const { getAllByText } = render(<SellProduct/>)

      await waitFor(() => {
        expect(getAllByText(/Product Name/).length).toBe(2)
      })
    });
  })
})