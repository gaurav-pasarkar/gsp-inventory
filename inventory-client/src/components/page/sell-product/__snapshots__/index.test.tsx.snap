// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sell products should render 1`] = `
<div>
  <div
    className="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-1 Mui-expanded WithStyles(ForwardRef(Accordion))-expanded-2 MuiPaper-elevation1"
  >
    <div
      aria-controls="panel1d-content"
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-5 Mui-expanded WithStyles(ForwardRef(AccordionSummary))-expanded-7"
      id="panel1d-header"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-6 Mui-expanded WithStyles(ForwardRef(AccordionSummary))-expanded-7"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Add Products
        </p>
      </div>
    </div>
    <div
      className="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-8"
    >
      <form
        className="makeStyles-root-3"
        onSubmit={[Function]}
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
          data-testid="productName"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              id="undefined-label"
            >
              Product Name
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Clear"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Clear"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-11"
                >
                  <span>
                    Product Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Add product
            </span>
          </button>
        </div>
      </form>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-1 MuiPaper-elevation1"
  >
    <div
      aria-controls="panel2d-content"
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-5"
      id="panel2d-header"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-6"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Collapsible Group Item #2
        </p>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel2d-header"
            id="panel2d-content"
            role="region"
          >
            <div
              className="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-8"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-1 MuiPaper-elevation1"
  >
    <div
      aria-controls="panel3d-content"
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-5"
      id="panel3d-header"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-6"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Collapsible Group Item #3
        </p>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel3d-header"
            id="panel3d-content"
            role="region"
          >
            <div
              className="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-8"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
